src/vmm/src/device_manager/mmio.rs

```diff
@@ -671,31 +665,27 @@ mod tests {
             // Use an exhaustive 'match' to make sure we cover all error variants.
             // When adding a new variant here, don't forget to also call this function with it.
             let msg = match err {
-                Error::Bus(_) => format!("{}{:?}", err, err),
+                Error::BusInsert(_) => format!("{}{:?}", err, err),
                 Error::Cmdline(_) => format!("{}{:?}", err, err),
                 Error::DeviceNotFound => format!("{}{:?}", err, err),
-                Error::EventFd(_) => format!("{}{:?}", err, err),
-                Error::IncorrectDeviceType => format!("{}{:?}", err, err),
+                Error::InvalidDeviceType => format!("{}{:?}", err, err),
                 Error::InternalDeviceError(_) => format!("{}{:?}", err, err),
-                Error::InvalidInput => format!("{}{:?}", err, err),
+                Error::InvalidIrqConfig => format!("{}{:?}", err, err),
                 Error::RegisterIoEvent(_) => format!("{}{:?}", err, err),
                 Error::RegisterIrqFd(_) => format!("{}{:?}", err, err),
-                Error::UpdateFailed => format!("{}{:?}", err, err),
-                Error::AllocatorError(_) => format!("{}{:?}", err, err),
+                Error::Allocator(_) => format!("{}{:?}", err, err),
             };
```

Comment:
I don't see a reason for this match, can this not be replaced like:
```suggestion
            let msg = format!("{err}{err:?}");
```

Reply:
Reason is mentioned in comment "// Use an exhaustive 'match' to make sure we cover all error variants.".

Reply:
I don't see the value in this, it doesn't add coverage.

Reply:
So this basically tests that error messages are not empty. Not so convinced in usefulness of this.
Also relying on the comment is not so great idea. It can be ignored easily.
Do we check that error message is not zero anywhere else? If someone will add new error with empty message it will be obvious without this test.

Anyway you can collapse this match to use "or":
```rust
    let msg = match err {
        Error::Bus(_)
        | Error::BusInsert(_)
        | Error::Cmdline(_)
        | Error::DeviceNotFound
        | Error::EventFd(_)
        | Error::IncorrectDeviceType
        | Error::InvalidDeviceType
        | Error::InternalDeviceError(_)
        | Error::InvalidInput
        | Error::InvalidIrqConfig
        | Error::RegisterIoEvent(_)
        | Error::RegisterIrqFd(_)
        | Error::UpdateFailed
        | Error::AllocatorError(_)
        | Error::Allocator(_) => format!("{err}{err:?}"),
    };

```

Reply:
This indeed might not bring a lot of value. We have lots of unit tests this way, unfortunately. If you feel like it, propose a change.
This PR is not related to this.

Reply:
I disagree that the PR is not related to removing this specific test. I tried tracing down the reason for these kind of tests, and they were indeed for code coverage reason: If we did not have these tests, then a _manually implemented_ `Display` or `Debug` implementation would not be covered. This is also why this match has to be exhaustive: If we added a match arm to the `Display` implementation, then we'd need to also add it to the test, otherwise the new arm would be uncovered. 

_However_ we have since switched to use `thiserror`, which removes manually implemented `Display` implementations. So the need for these tests that "game the metric" is no longer present. For this reason, I would argue that removing them should happen in the same commit that introduces a `thiserror` implementation. Sadly this hasn't happened so far :(

Reply:
TBH I do not know how `thiserror` works behind the scenes, but was wondering why you say that this test was only relevant for manual implementation and is not for `thiserror`? Will the lines adding the "#[error("....")]" display message get covered automatically with `thiserror`?

Reply:
The lines marked as `#[error(...)]` are not really lines of code, they are instructions for the `thiserror` proc-macro. They are treated by code coverage tooling just like comments: neither covered nor uncovered. See the below screenshot of a coverage report from an enum using `thiserror` from the api_server crate:
![image](https://user-images.githubusercontent.com/115481277/213250384-f3820bc1-3542-4fdd-be78-adf4e3df94f6.png)

Reply:
thanks @roypat for clarifying. I will remove the test.